azkaban-exec-server/src/main/java/azkaban/execapp/JobRunner.java

```diff
@@ -626,6 +622,14 @@ private void doRun() {
     writeStatus();
   }
 
+  private void uploadExecutableNode() {
+    try {
+      this.loader.uploadExecutableNode(this.node, this.props);
+    } catch (final ExecutorManagerException e1) {
```

Comment:
Yes, I have the same question. If in doubt, I am in favor of failing early. At least we understand the behavior better.

That said, I would suggest that we make small incremental changes. e.g. fix this issue first and then do further improvements.